examples/gno.land/p/moul/authz/authz.gno

```diff
@@ -0,0 +1,248 @@
+// Package authz provides flexible authorization control for privileged actions.
+//
+// # Authorization Strategies
+//
+// The package supports multiple authorization strategies:
+//   - Member-based: Single user or team of users
+//   - Contract-based: Async authorization (e.g., via DAO)
+//   - Auto-accept: Allow all actions
+//   - Drop: Deny all actions
+//
+// Core Components
+//
+//   - Authority interface: Base interface implemented by all authorities
+//   - Authorizer: Main wrapper object for authority management
+//   - MemberAuthority: Manages authorized addresses
+//   - ContractAuthority: Delegates to another contract
+//   - AutoAcceptAuthority: Accepts all actions
+//   - DroppedAuthority: Denies all actions
+//
+// Quick Start
+//
+//	// Initialize with contract deployer as authority
+//	var auth = authz.New()
+//
+//	// Create functions that require authorization
+//	func UpdateConfig(newValue string) error {
+//	    return auth.Do("update_config", func() error {
+//	        config = newValue
+//	        return nil
+//	    })
+//	}
+//
+// See example_test.gno for more usage examples.
+package authz
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/avl/rotree"
+	"gno.land/p/demo/ufmt"
+	"gno.land/p/moul/addrset"
+	"gno.land/p/moul/once"
+)
+
+// Authorizer is the main wrapper object that handles authority management
+type Authorizer struct {
+	current Authority
+}
+
+// Authority represents an entity that can authorize privileged actions
+type Authority interface {
+	// Authorize executes a privileged action if the caller is authorized
+	Authorize(title string, action PrivilegedAction) error
```

Comment:
Wondering if it makes sense to optionally send arguments to the authority 🤔, for example as:

```suggestion
	Authorize(title string, action PrivilegedAction, args ...interface{}) error
```

Reason for this would be a case where Authority creates a proposal to execute the privileged action; It would be useful to be able to list parameters for example.

I'm not sure but if arguments are supported we provably could consider and explore a different approach for `boards2` so we can use this package there.

Reply:
I know the idea if this package is to be minimal but I though it might be worth bringing this up

Reply:
https://github.com/gnolang/gno/pull/3700/commits/dd98090019bb52492fee97585fbaadf5ce652be5